crates/hathor-songs/src/database.rs

```diff
@@ -15,6 +23,38 @@ pub fn get_connection(db_path: &Path) -> Result<Connection, Box<dyn std::error::
     Ok(conn)
 }
 
+pub fn query_map_to_audiofile<ParamType>(
+    conn: &Connection,
+    sql: &str,
+    parameters: ParamType,
+) -> Result<Vec<AudioFile>, Box<dyn Error>>
+where
+    ParamType: Params,
+{
+    Ok(conn
+        .prepare(sql)?
+        .query_map(parameters, |row| {
+            // Literally impossible to return an error variant here.
+            // So instead we just unwrap and will panic and crash everything lol.
+            // TODO: Fix this dog water.
+            let song_path = PathBuf::from_str(&row.get::<_, String>(7)?).unwrap();
+            let img_path = PathBuf::from_str(&row.get::<_, String>(8)?).unwrap();
```

Comment:
@vablings Any idea how to address the comments here? I can't figure out how to return an error variant here if the path doesn't exist instead of unwrapping. Essentially at the moment, if a song exists in the DB and then the song gets moved/deleted, this query will panic rather than returning an error variant/skipping the missing entries.

Reply:
'PathBuf' doesn't check if the Path is valid, it's essentially just a wrapper to String/OsString

[Here is the implementation ](https://doc.rust-lang.org/src/std/path.rs.html#1738)

Assuming your string is already valid it should never fail to convert to from String to PathBuf hence why it's infallible 

I think the best approach is to validate the string (probably already happens by virtue of it being some kind of concrete type from the database. I just can't see the rest because I'm on mobile) and then verify the path with '.exists()'

Reply:
Didn't realise that. In that case I think it's fine to unwrap here as there's no chance it can fail.